### PR TITLE
Task 4 issue #8

### PR DESCRIPTION
**Pull request by DragonTamer9001**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -2425,9 +2434,44 @@ void handle_movement(double dt) {
...
+        }
+        else if (autoWalkState == GLFW_RELEASE) {
+        	*allow_next_autowalk_press_ptr = true;
+        }//end auto walk
+        	
     }
     float vx, vy, vz;
-    get_motion_vector(g->flying, sz, sx, s->rx, s->ry, &vx, &vy, &vz);
+    
+    //New system overrides this code piece
```
Comment (XJamesWrightX) [inline src/main.c:2465]: Having a reason for commenting out code is alright. However according to Uncle Bob, you should really not do this in order to keep the code clean. If this note is really important, I suggest you document it elsewhere & not in the program itself. Same goes for all the other commented out code in the program. I do see on line 2455 as of now there is also commented out code of which does not have a reason & I suggest just removing that then if it has no importance.

**Review by XJamesWrightX — CHANGES_REQUESTED** [review]

File: src/main.c [hunk 2]
```diff
@@ -2408,11 +2411,17 @@ void handle_mouse_input() {
     }
 }
 
-void handle_movement(double dt) {
+
+void handle_movement(double dt, bool *is_auto_walking_ptr, bool *allow_next_autowalk_press_ptr, bool *do_auto_walk_ptr) {
```
Comment (lbrennaman) [inline src/main.c:2415]: I would recommend changing the three parameters you added into a struct containing those variables to make the code cleaner.

**Review by lbrennaman — CHANGES_REQUESTED** [review]

File: src/main.c (hunk 1 above)
Comment (DragonTamer9001) [reply src/main.c]: Done. Thanks for the feedback!

File: src/main.c (hunk 2 above)
Comment (DragonTamer9001) [reply src/main.c:2415]: I would do this, but compared to some other changes in master by other members, most use a similar variable set-up like mine.

**Pull request merged**